src/ReverseProxy/Health/ActiveHealthCheckMonitor.cs

```diff
@@ -29,10 +29,14 @@ public ActiveHealthCheckMonitor(
         TimeProvider timeProvider,
         ILogger<ActiveHealthCheckMonitor> logger)
     {
-        _monitorOptions = monitorOptions?.Value ?? throw new ArgumentNullException(nameof(monitorOptions));
-        _policies = policies?.ToDictionaryByUniqueId(p => p.Name) ?? throw new ArgumentNullException(nameof(policies));
-        _probingRequestFactory = probingRequestFactory ?? throw new ArgumentNullException(nameof(probingRequestFactory));
-        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        ArgumentNullException.ThrowIfNull(monitorOptions?.Value);
+        _monitorOptions = monitorOptions?.Value;
```

Comment:
Unnecessary null propagation usage since we already have null check before
```suggestion
        _monitorOptions = monitorOptions.Value;
```